### DISCUSSION TITLE
Method to get current year?

### DISCUSSION THREAD

**Discussion by ecupaio** [maizzle/framework#763] [Q&A] [answered]
I'm attempting to display a copyright year on my email template but I'm unsure of a method where I can get the date without using javascript. If I write a script it gets compiled to the production template instead of printing the value so I'm wondering if there's something in the configs I can just pull out and place as a variable like I did with the frontmatter in the templates.

**Comment by michellaurent** [comment, marked as answer]
I think that you can use [a function](https://maizzle.com/docs/functions) for that.

In config.js:
```js
current_year: function() {
    return new Date().getFullYear()
}
```

In your template:
```
<p>{{ page.current_year }}</p>

**Reply by cossssmin** [reply]
Yup that's how I do it too 👍

You can even take it a step further and remove the need for `page.` in the expression by defining the method as a top-level variable - https://maizzle.com/docs/environments#top-level-variables

```js
module.exports = {
  locals: {
    current_year: function() {
      return new Date().getFullYear()
    }
  }
}
```

You'd then just do:

```html
<p>{{ current_year }}</p>
```

**Reply by ecupaio** [reply]
noice! I'm adding page.current year to my layout file and it works. What would be the use case of a top level variable?

**Reply by michellaurent** [reply]
@ecupaio The only advantage is that you don't have to prefix the variable name with `page.`

**Reply by cossssmin** [reply]
Mostly if you don't want to write `page.` all the time, or if you somehow really need to separate data within them.

**Reply by ecupaio** [reply]
got it. thanks for the tip!

**Comment by ecupaio** [comment]
oh nice! This works. To provide the [wisdom of the ancients](https://xkcd.com/979/), here's exactly what I did to achieve this: 
In config.js, I added the current_year function to the module.exports object.: 
config.js
```
module.exports = {
  build: {
    templates: {
      source: 'src/templates',
      destination: {
        path: 'build_local',
      },
      assets: {
        source: 'src/images',
        destination: 'images',
      },

    },
  },
  current_year: function() {
      return new Date().getFullYear()
  }
}
```

Then in my layout I added the following code: 
```
<p>{{ page.current_year }}</p>
```

You can add this to your template files as well.

**Comment by michellaurent** [comment]
You don't have to do it in both config files. They are merged together.

**Reply by ecupaio** [reply]
so just inside the config file yeah?

**Reply by michellaurent** [reply]
Yes.

**Reply by ecupaio** [reply]
excellent, i updated my reply with the correct steps.
